src/uct/tcp/tcp_sockcm.h

```diff
@@ -7,30 +7,39 @@
 #include <uct/base/uct_cm.h>
 
 
+typedef struct uct_tcp_sockcm_ep   uct_tcp_sockcm_ep_t;
+
+
 typedef struct uct_tcp_sa_arg {
-    int             fd;
-    ucs_list_link_t list;
+    int                 fd;
+    ucs_list_link_t     list;
+    uct_tcp_sockcm_ep_t *ep;
```

Comment:
let's remove this struct and use uct_tcp_sockcm_ep_t directly instead as we discussed
(put in on the list etc.)

Reply:
what do you mean? this is the context for the async handler. how can i access the ep without it from the handler?